cmd/openshift-install/internal_integration_test.go

```diff
@@ -271,6 +335,8 @@ func ignitionStorageCmp(ts *testscript.TestScript, neg bool, args []string) {
 	eData, err := os.ReadFile(eFilePathAbs)
 	ts.Check(err)
 
+	eData = replaceReleaseImage(string(eData))
+	aData = replaceReleaseImage(string(aData))
```

Comment:
I would not have expected that `aData` needs to be modified.
We have to modify `eData` because the expected release image can't be known in advance when creating the static file. But the reason we need that is because `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` modifies what is actually output.
If we force both to the same value before comparing them, then it will always be the same and we aren't actually testing our code.
We're looking for the placeholder `$RELEASE_IMAGE` though, so I'd assume this line has no effect?

Reply:
Untill now, even in the CI, we were always comparing the `$RELEASE_VERSION` against the [version](https://github.com/openshift/installer/blob/master/pkg/asset/releaseimage/default.go#L23) hardcoded in the code i.e. we were not running the tests against a CI release version.  This also means we were setting the same release version in the actual and expected test data here https://github.com/openshift/installer/blob/275af57895a88f9df7dc392ca87fc0d51418b1be/cmd/openshift-install/testdata/agent/unconfigured-ignition/configurations/from-ztp-manifests.txt#L47-L63

We can not change the `defaultOriginalReleaseVersion` in the code for the purpose of running the tests in CI env, so it is good to change/overwrite the actual and expected test data so that like before `RELEASE_IMAGE` in actual and expected files contains the  actual value of the `RELEASE_IMAGE` used in the CI env against which the tests are running and not always the `defaultOriginalReleaseVersion`